tests/ui/test_ui_tools.py

```diff
@@ -1454,6 +1454,14 @@ class TestMsgInfoView:
     @pytest.fixture(autouse=True)
     def mock_external_classes(self, mocker, monkeypatch, message_fixture):
         self.controller = mocker.Mock()
+        self.controller.model.index = {
+            # These message IDs are from message_fixture. Patch
+            # show_edit_history_label separately wherever required.
+            'edited_messages': set((537286, 537287, 537288)),
+        }
+        self.controller.model.initial_data = {
+            'realm_allow_edit_history': True,
```

Comment:
Since this and the patch above it are required for other tests in `TestMsgInfoView` as well, I only patched them once here. Though, I have simplified these in the latest update.

I am not sure if we can overwrite a fixture patch (this particular fixture, `mock_external_classes`, is an `autouse` fixture which is used everywhere in the class).

Besides, the `edit_history` test simply cares about the `show_edit_history_label` value (and how that value is set [with `and` between booleans] is trivial in comparison).